packages/zoltan2/core/src/algorithms/color/Zoltan2_AlgHybrid2GL.hpp

```diff
@@ -0,0 +1,1532 @@
+                                 Kokkos::View<lno_t*,
+				              device_type, 
+					      Kokkos::MemoryTraits<Kokkos::Atomic> > verts_to_recolor_atomic,
+				 Kokkos::View<int*, 
+				              device_type, 
+					      Kokkos::MemoryTraits<Kokkos::Atomic>> verts_to_recolor_size_atomic,
+				 Kokkos::View<lno_t*,
+				              device_type,
+					      Kokkos::MemoryTraits<Kokkos::Atomic> > verts_to_send_atomic,
+				 Kokkos::View<size_t*, 
+				              device_type, 
+					      Kokkos::MemoryTraits<Kokkos::Atomic>> verts_to_send_size_atomic,
+			         Kokkos::View<gno_t*, device_type> recoloringSize,
+				 Kokkos::View<int*,
+				              device_type> rand,
+			         Kokkos::View<gno_t*,
+				              device_type> gid,
+                                 Kokkos::View<gno_t*,
+				              device_type> ghost_degrees,
+			         bool recolor_degrees) = 0;
+    
+    //Entry point for serial conflict detection
+    virtual void detectConflicts_serial(const size_t n_local,
+		                 typename Kokkos::View<offset_t*, device_type >::HostMirror dist_offsets_host,
+		                 typename Kokkos::View<lno_t*, device_type >::HostMirror dist_adjs_host,
+				 typename Kokkos::View<int*,device_type >::HostMirror femv_colors,
+				 typename Kokkos::View<lno_t*, device_type >::HostMirror boundary_verts_view,
+				 gno_t boundary_size,
+                                 typename Kokkos::View<lno_t*,device_type>::HostMirror verts_to_recolor_atomic,
+				 typename Kokkos::View<int*,device_type>::HostMirror verts_to_recolor_size_atomic,
+				 typename Kokkos::View<lno_t*,device_type>::HostMirror verts_to_send_atomic,
+				 typename Kokkos::View<size_t*,device_type>::HostMirror verts_to_send_size_atomic,
+				 typename Kokkos::View<gno_t*, device_type>::HostMirror recoloringSize,
+			         typename Kokkos::View<int*,  device_type>::HostMirror rand,
+			         typename Kokkos::View<gno_t*,device_type>::HostMirror gid,
+                                 typename Kokkos::View<gno_t*,device_type>::HostMirror ghost_degrees,
+				 bool recolor_degrees) = 0;
+    //Entry point for the construction of the boundary
+    //INPUT ARGS
+    //  n_local is the number of vertices owned by the current process
+    //
+    //  dist_offsets_dev is the device view that holds the CSR offsets
+    //
+    //  dist_adjs_dev is the device view that holds CSR adjacencies
+    //
+    //  dist_offsets_host is the hostmirror that holds the CSR offsets
+    //   
+    //  dist_adjs_host is the hostmirror that holds the CSR adjacencies
+    //
+    //OUTPUT ARGS
+    //  boundary_verts is an unallocated device view that will hold the
+    //                 list of boundary vertices.
+    //
+    //  boundary_size will hold the size of the list of boundary vertices
+    //
+    //  verts_to_send_atomic will hold the list of vertices to send
+    //
+    //  verts_to_send_size_atomic will hold the size of the list of
```

Comment:
Does verts_to_send_size_atomic differ from verts_to_send_atomic.extent(0)?  Also, say why it is a view rather than a size_t.